### PR TITLE
Feature/ms 67 checkout logged user

### PR DESCRIPTION
**Pull request by andrystylist**

### CODE REVIEW

File: src/components/Admin/Products/ProductForm.jsx [hunk 1]
```diff
@@ -1,10 +1,207 @@
...
+      .test(
+        'fileSize',
+        'File too large',
+        value => {
+          console.log('--- AC: ', {
+            value,
+            size: value?.size,
+            FILE_SIZE,
+            isValid: value?.size <= FILE_SIZE
+          })
```
Comment (ht1204) [inline src/components/Admin/Products/ProductForm.jsx:49]: remove this log, thx.

File: src/components/Admin/Products/ProductTable.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
+/* eslint-disable react/prop-types */
+import React from 'react'
+import { GoPencil } from 'react-icons/go'
+import Loading from '../../Loading/Loading'
+import './ProductTable.scss'
+
+function ProductTable ({ products, isLoading, onOpenEdit }) {
+  const headboard = ['Name', 'Description', 'Code', 'Category', 'Price', 'Stock', '']
+  const isListEmpty = !products || products.length === 0
```
Comment (ht1204) [inline src/components/Admin/Products/ProductTable.jsx:9]: `const isListEmpty = !products || !products.length`

File: src/components/Admin/Products/UploadFile.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable react/prop-types */
+import React from 'react'
+
+function UploadFile ({ setFieldValue, ...rest }) {
+  const handleOnChange = (e) => {
+    // Object is possibly null error w/o check
```
Comment (ht1204) [inline src/components/Admin/Products/UploadFile.jsx:6]: remove this comment

File: src/components/Modal/ControlledModal.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  const stopPropagation = (event) => {
+    event.stopPropagation()
+  }
+
+  const handleClose = () => {
+    onClose()
+  }
+
+  useEffect(() => {
+    const handleScape = (event) => {
```
Comment (ht1204) [inline src/components/Modal/ControlledModal.jsx:21]: `const handleEscape = (event) => {`

File: src/components/Modal/ControlledModal.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    // Función para habilitar el desplazamiento del documento cuando el modal se cierra
+    const enableScroll = () => {
+      document.body.style.overflow = 'auto'
+    }
+
+    if (isOpen) { // Se abrió el modal
+      disableScroll()
+    } else { // Se cerró el modal
```
Comment (ht1204) [inline src/components/Modal/ControlledModal.jsx:41]: remove unnecessary comments, thx.

File: src/hooks/useAddProductToCart.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  const handleAddToCart = (product, quantity) => {
+    setProductsList(prevProducts => {
+      const doesProductExist = prevProducts.find((productCart) => productCart.id === product.id)
+
+      if (!doesProductExist) {
+        return [...prevProducts, { ...product, quantity }]
+      }
+
+      // If this reduce is ran the product already exists
```
Comment (ht1204) [inline src/hooks/useAddProductToCart.js:15]: remove this comment

File: src/pages/Cart/Cart.jsx [hunk 7]
```diff
@@ -55,7 +57,11 @@ function Cart () {
 
   const handleSubmitOrder = (event) => {
     event.preventDefault()
-    // TODO: Navegar programáticamente a Checkout si todo está bien
+    console.log(authData)
```
Comment (ht1204) [inline src/pages/Cart/Cart.jsx:60]: remove this log.

File: src/pages/Checkout/Checkout.jsx [hunk 8]
```diff
@@ -70,21 +79,49 @@ function Checkout () {
...
+
+  useEffect(() => {
+    return () => {
+      if (order) {
+        setProductsList([])
+      }
     }
+  }, [])
+
+  if (productsList.length === 0) {
```
Comment (ht1204) [inline src/pages/Checkout/Checkout.jsx:123]: `if (!productsList.length) {`

**Review by ht1204 — COMMENTED** [review]
Please check the comments to address and approve, thx.

**Review by ht1204 — APPROVED** [review]

File: src/components/Admin/Products/ProductTable.jsx (hunk 2 above)
Comment (andrystylist) [reply src/components/Admin/Products/ProductTable.jsx:9]: I will make the change although I see it with more meaning this way but I will make this change anyway :)

File: src/pages/Checkout/Checkout.jsx (hunk 8 above)
Comment (andrystylist) [reply src/pages/Checkout/Checkout.jsx:123]: done

**Pull request merged**